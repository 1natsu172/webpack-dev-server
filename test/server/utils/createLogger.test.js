'use strict';

const webpack = require('webpack');
const createLogger = require('../../../lib/utils/createLogger');
const config = require('../../fixtures/simple-config/webpack.config');

describe('createLogger util', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('without the compiler', () => {
    let consoleMock;

    beforeEach(() => {
      consoleMock = jest.spyOn(console, 'info');
    });

    it('should output log when use the default level', () => {
      const logger = createLogger(null);

      logger.info('foo');

      expect(consoleMock).toBeCalled();
      expect(consoleMock.mock.calls[0][0]).toMatchSnapshot();
    });

    it('should not output log when specify the level', () => {
      const logger = createLogger(null, 'none');

      logger.info('foo');
      expect(consoleMock).not.toBeCalled();
    });
  });

  describe('with the compiler', () => {
    let consoleMock;

    beforeEach(() => {
      consoleMock = jest.spyOn(process.stderr, 'write');
    });

    it('should output log when use the default level', () => {
      const compiler = webpack(config);
      const logger = createLogger(compiler);

      logger.info('foo');
      expect(consoleMock).toBeCalled();
      expect(consoleMock.mock.calls[0][0]).toMatchSnapshot();
    });

    it('should not output log when specify the level', () => {
      const compiler = webpack({
        ...config,
        infrastructureLogging: {
          level: 'none',
        },
      });
      const logger = createLogger(compiler);

      logger.info('foo');
      expect(consoleMock).not.toBeCalled();
    });
  });
});
