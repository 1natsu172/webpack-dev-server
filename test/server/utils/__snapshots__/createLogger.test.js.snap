// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createLogger util with the compiler should output log when use the default level 1`] = `
"<i> [webpack-dev-server] foo
"
`;

exports[`createLogger util without the compiler should output log when use the default level 1`] = `"[webpack-dev-server] foo"`;
